untitled1/src/main/com/aca/homework/week6/myinteger/MyInteger.java

```diff
@@ -0,0 +1,29 @@
+package com.aca.homework.week6.myinteger;
+
+public class MyInteger {
+    private static MyInteger[] myIntegers = new MyInteger[201];
+    private int value;
+
+    public MyInteger(int value) {
+        this.value = value;
+    }
+
+    public int getValue() {
+        return value;
+    }
+
+    public static MyInteger valueOf(int value) {
+        if (isValueOutOfRange(value)) {
+            System.out.println(value + " is out of range [-100,100] we can not cache it  with number in this range,needed another range for caching int values");
+            return null;
```

Comment:
Can i throw exception in this case,It's not so clear to me in what case to  throw  exception , in what case to return null․In very few cases, it is clear which one to choose